### PR TITLE
Build branch filter

### PR DESCRIPTION
**Pull request by cadeParade**
This feature adds a purely-frontend implementation of a branch filter.

It takes all the builds that are loaded in the store and adds them to a dropdown.

It is not a perfect solution for all cases, but I estimate that it is a pretty good solution for at least 80% of cases.
The caveats are related to pagination and infinity loading. There's currently no way to request builds from a specific branch in the API, plus there would be many questions raised by doing that. For instance, which branches should we load? Should we load all branches from the history of the repo for the dropdown? What about pagination? Should we paginate a filtered view? 

I've gotten around this a bit by allowing infinity loader to work when all branches are selected, and when the branches are filtered, it does not load more builds, it just shows the builds that are already in the store that are for that branch.

### CODE REVIEW

File: app/components/project-container.js [hunk 1]
```diff
@@ -37,7 +44,22 @@ export default Component.extend(PollingMixin, {
     return utils.sortAndCleanBuilds(filteredBuilds);
   }),
 
+  selectedBranch: allBranchesString,
+  projectBranches: computed('builds.@each.branch', function() {
+    return [allBranchesString].concat(Array.from(new Set(this.get('builds').mapBy('branch'))));
```
[the remark this answers is not in the document]
Comment (cadeParade) [reply app/components/project-container.js]: 😁

[the remark this answers is not in the document]
Comment (cadeParade) [reply app/components/project-container.js]: ok

File: app/templates/components/project-container.hbs [hunk 2]
```diff
@@ -60,13 +74,17 @@
...
       {{/each}}
-      {{#if canLoadMore}}
+      {{#if shouldLoadMore}}
         {{#infinity-loader infinityModel=infinityBuilds}}
           {{loading-build-card}}
         {{/infinity-loader}}
+      {{else}}
+        <div class="p-4 bg-white border-b flex justify-center text-gray-400">
+            To load more builds, select "All branches" from the select menu above
+        </div>
```
[the remark this answers is not in the document]
Comment (cadeParade) [reply app/templates/components/project-container.hbs]: yes, excellent point.

File: app/components/project-container.js [hunk 3]
```diff
@@ -37,7 +44,24 @@ export default Component.extend(PollingMixin, {
     return utils.sortAndCleanBuilds(filteredBuilds);
   }),
 
+  selectedBranch: allBranchesString,
+  projectBranches: computed('builds.@each.branch', function() {
+    const allBranches = this.get('builds').mapBy('branch');
+    const uniqueBranches = Array.from(new Set(allBranches));
+    return [allBranchesString].concat(uniqueBranches);
```
Comment (maprules1000) [inline app/components/project-container.js]: Wayyyy more readable

**Comment by cadeParade** [comment]
The dropdowns are still enabled, they just have a default gray background color now. cc @PixelJanitor . I'm going to merge for now, and we can iterate on power-select styles in future PRs.

**Pull request merged**